### PR TITLE
add modifier for handling the pixel CPE for HLT phase2

### PR DESCRIPTION
**Pull request by mtosi**
modifier to be used for HLT TDR, in order to make sure that we are using the generic CPE (w/ hardcoded CPE error in the seeding step, and generic error in the fitter when the track angle is available)

I'm going to make a new PR w/ the backport to 111X

@fwyzard

### CODE REVIEW

**Comment by mtosi** [comment]
@hevjinyarar

**Comment by mtosi** [comment]
in 111X we will need the specific cff file, though

File: RecoTracker/TransientTrackingRecHit/python/TTRHBuilderWithTemplate_cfi.py [hunk 1]
```diff
@@ -18,3 +18,7 @@
 # Turn off template reco for phase 2 (when not supported)
 from Configuration.ProcessModifiers.phase2_PixelCPEGeneric_cff import phase2_PixelCPEGeneric
 phase2_PixelCPEGeneric.toModify(TTRHBuilderAngleAndTemplate, PixelCPE = 'PixelCPEGeneric')
+
+# Turn off template reco for phase 2 HLT
+from Configuration.ProcessModifiers.phase2_PixelCPEGeneric4HLT_cff import phase2_PixelCPEGeneric4HLT
+phase2_PixelCPEGeneric4HLT.toModify(TTRHBuilderAngleAndTemplate, PixelCPE = 'PixelCPEGeneric')
```
Comment (slava77) [inline RecoTracker/TransientTrackingRecHit/python/TTRHBuilderWithTemplate_cfi.py:24]: how is this modifier expected to differ from the existing ```phase2_PixelCPEGeneric``` ?

**Review by slava77 — COMMENTED** [review]
a non-HLT variant apparently exists already.
It also includes the following change in RecoLocalTracker/SiPixelRecHits/python/PixelCPEGeneric_cfi.py

```python
phase2_PixelCPEGeneric.toModify(PixelCPEGenericESProducer,
  UseErrorsFromTemplates = False,    # no GenErrors
  LoadTemplatesFromDB = False,       # do not load templates
)
```

is this not relevant for what's proposed in this PR? (would it harm?)

Is there a non-HLT use case for what's called ```phase2_PixelCPEGeneric4HLT``` in this PR?
If so, some better name may be in order.

**Comment by slava77** [comment]
> in 111X we will need the specific cff file, though

please elaborate.
Thank you.

**Comment by slava77** [comment]
> Is there a non-HLT use case for what's called `phase2_PixelCPEGeneric4HLT` in this PR?
> If so, some better name may be in order.

alternatively, can this become a more generic modifier for a-la HLT setups?
IIUC, once the HLT configs are in the confDb (or in a dump fragment file), this modifier will have no impact; is this correct?

**Comment by fwyzard** [comment]
I think in any release we will actually need a cfi/cff for the HLT
configuration, rather than a modifier ?

File: RecoTracker/TransientTrackingRecHit/python/TTRHBuilderWithTemplate_cfi.py (hunk 1 above)
Comment (mmusich) [reply RecoTracker/TransientTrackingRecHit/python/TTRHBuilderWithTemplate_cfi.py:24]: As you already pointed out above this modifier differs from the existing one in the fact that this one just switches all the template calls into generic calls, while the existing one also removes any usage of template info (e.g. for the error assignment).

Comment (slava77) [reply RecoTracker/TransientTrackingRecHit/python/TTRHBuilderWithTemplate_cfi.py:24]: Is this the intended behavior?

... anyways, from https://github.com/cms-sw/cmssw/pull/30465#issuecomment-652076405 it sounds like the expected solution for HLT will not use a modifier

Comment (mmusich) [reply RecoTracker/TransientTrackingRecHit/python/TTRHBuilderWithTemplate_cfi.py:24]: > Is this the intended behavior?

yes. In the case I provided earlier (in PR https://github.com/cms-sw/cmssw/pull/29018 ) - as you can check in the presentation I gave at the Reco/AT meeting [*],  we don't have (yet) GenErrors for the 3D sensors, so *all* template information must be avoided. Here, instead, @mtosi  just wants to switch Template with Generic CPE (standard configuration)

[*] https://indico.cern.ch/event/893830/contributions/3770697/ 

> ... anyways, from #30465 (comment) it sounds like the expected solution for HLT will not use a modifier

right. I would propose to close this PR.

Comment (slava77) [reply RecoTracker/TransientTrackingRecHit/python/TTRHBuilderWithTemplate_cfi.py:24]: > right. I would propose to close this PR.

@mtosi 
do you agree?

Comment (slava77) [reply RecoTracker/TransientTrackingRecHit/python/TTRHBuilderWithTemplate_cfi.py:24]: > > right. I would propose to close this PR.
> 
> @mtosi
> do you agree?

ping

**Comment by mtosi** [comment]
yes, we need the cff for the HLT configuration

**Pull request closed without merging**